code/obj/machinery/computer/bank_records.dm

```diff
@@ -264,6 +266,35 @@
 					if (transto == "Payroll") wagesystem.station_budget += amount
 					if (transto == "Shipping") wagesystem.shipping_budget += amount
 					if (transto == "Research") wagesystem.research_budget += amount
+				else if(href_list["bonus"])
+					if(!(world.time >= src.bonus_rate_limit_time))
+						boutput(usr, SPAN_ALERT("NT Regulations forbid issuing multiple staff incentives within five minutes."))
+						return
+
+					var/amount = input(usr, "How many credits should we issue to each staff member?", "Issue Bonus", 100) as null|num
+					if(isnull(amount)) return
+
+					amount = ceil(clamp(amount, 1, 999999))
+					var/bonus_amt = (length(data_core.bank.records) * amount)
+					if ( bonus_amt > wagesystem.station_budget)
+						boutput(usr, SPAN_ALERT("Total bonus cost would be [bonus_amt][CREDIT_SIGN], payroll budget is only [wagesystem.station_budget][CREDIT_SIGN]!"))
```

Comment:
This needs to be checked right before spending it. Right now someone could open the UI past this, to the message section, and drain all the accounts, then have negative money